### PR TITLE
Add user_attrs argument to study.enqueue_trial

### PR DESCRIPTION
**Pull request by knshnb**
## Motivation
Address
> [updated:2021/11/28] we decide to add user_attrs parameter to enqueue_trial

of #2955

## Description of the changes
I added `user_attrs` argument to `study.enqueue_trial()` as well as the corresponding explanation in docstring and tests.

### CODE REVIEW

File: optuna/study/study.py [hunk 1]
```diff
@@ -846,23 +848,30 @@ def objective(trial):
...
 
                 assert study.trials[0].params == {"x": 5}
                 assert study.trials[1].params == {"x": 0}
+                assert study.trials[1].user_attrs == {"memo": "optimal"}
 
         Args:
             params:
                 Parameter values to pass your objective function.
+            user_attrs:
+                A dictionary of user specific attributes other than params.
```
Comment (nzw0301) [inline optuna/study/study.py:862]: how about this?

```suggestion
                A dictionary of user-specific attributes other than ``params``.
```

**Review by nzw0301 — CHANGES_REQUESTED** [review]
Great. Could you look at my comment?

**Comment by knshnb** [comment]
Thank you! Updated.

**Review by nzw0301 — APPROVED** [review]
LGTM! thanks!

**Review by himkt — APPROVED** [review]
LGTM!

**Pull request merged**